lib/plugins/print/print.js

```diff
@@ -91,9 +104,44 @@ class Print {
         this.serverless.variables.service = service;
         return this.serverless.variables.populateObject(service)
           .then((populated) => {
-            this.strip(populated);
-            const out = JSON.parse(jc.stringify(populated));
-            this.serverless.cli.consoleLog(YAML.dump(out, { noRefs: true }));
+            let conf = populated;
+            this.strip(conf);
+
+            // dig into the object
+            if (this.options.path) {
+              const steps = this.options.path.split('.');
+              for (const step of steps) {
```

Comment:
I'm traversing an object in a synchronous way, I'm a bit confused why you would want to make that operation behave asynchronously? Are you saying this project does not allow any for-loops inside of functions that happen to return a promise?

Reply:
For now, I will change it to be `steps.forEach()`, which is more consistent with the rest of the codebase.

Reply:
Thank you for response 👍 
Yes, I know there are a lot of synchronous loops in promises in our codebase.
However, we(the framework maintainer team) have decided to use asynchronous way before as possible as we can, because current unit tests have sometimes failed with an unknown cause, and we are considering the cause is that haven't caught errors properly because of mixing synchronous/asynchronous. 

What do you think?

Reply:
Maybe you can show me what kind of problems you have seen? Maybe I can help.

Reply:
@ronkorving 
Thank you for validating the behavior 👍 

>Maybe you can show me what kind of problems you have seen? 

Unit tests of the framework have been flaky for a long time, i.e. it fails sometimes with uncaught or timeout errors but it fixes with re-building Travis again in many cases. 
https://travis-ci.org/serverless/serverless/jobs/312633967
https://travis-ci.org/serverless/serverless/builds/393374487

We have found some place which don't handle promise error, and maybe this leads the flaky tests, but it is hard to remove it all places.
Therefore we recently unify it with BbPromse as possible as we can.

Reply:
The timeout seems to be set to 2 seconds, and if I/O is slow (which it can be, especially on a free service like TravisCI), maybe it's just a matter of increasing the timeout value to something a bit higher? The tests that fail in the job you linked to are the exact same 2 tests in Node 4 and 6. Is it always those 2 that time out?

https://mochajs.org/#timeouts